### ISSUE TITLE
Resolve FileNotFoundException when browsing

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#549]
Investigate and fix the following exception related to browsing the directory of a Maven 2 repository:
```
22:42:07.754 28-01-2018 | ERROR | qtp1914190643-20     | o.c.s.providers.repository.ProxyRepositoryProvider | Failed to resolve Path for prixied artifact [storage-common-proxies]/[carlspring]/[.index/]
org.carlspring.strongbox.storage.ArtifactStorageException: java.io.FileNotFoundException: The artifact path is a directory: [/strongbox/strongbox-vault/storages/storage-common-proxies/carlspring/.index]
	at org.carlspring.strongbox.services.ArtifactManagementService.store(ArtifactManagementService.java:124)
	at org.carlspring.strongbox.services.ArtifactManagementService$$FastClassBySpringCGLIB$$b290f309.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:747)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89)
	at org.carlspring.strongbox.data.tx.OEntityUnproxyAspect.transactional(OEntityUnproxyAspect.java:56)
	at sun.reflect.GeneratedMethodAccessor54.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:643)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:632)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:294)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:98)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
	at org.carlspring.strongbox.services.ArtifactManagementService$$EnhancerBySpringCGLIB$$6fd15f58.store(<generated>)
	at org.carlspring.strongbox.providers.repository.proxied.LocalStorageProxyRepositoryArtifactResolver.onSuccessfulProxyRepositoryResponse(LocalStorageProxyRepositoryArtifactResolver.java:72)
	at org.carlspring.strongbox.providers.repository.proxied.ProxyRepositoryArtifactResolver.getInputStream(ProxyRepositoryArtifactResolver.java:107)
	at org.carlspring.strongbox.providers.repository.ProxyRepositoryProvider.resolvePathForceFetch(ProxyRepositoryProvider.java:146)
	at org.carlspring.strongbox.providers.repository.ProxyRepositoryProvider.resolvePath(ProxyRepositoryProvider.java:137)
	at org.carlspring.strongbox.providers.repository.ProxyRepositoryProvider.resolvePath(ProxyRepositoryProvider.java:37)
	at org.carlspring.strongbox.providers.repository.GroupRepositoryProvider.resolvePathFromGroupMemberOrTraverse(GroupRepositoryProvider.java:164)
	at org.carlspring.strongbox.providers.repository.GroupRepositoryProvider.resolvePathTraversal(GroupRepositoryProvider.java:110)
	at org.carlspring.strongbox.providers.repository.GroupRepositoryProvider.resolvePathFromGroupMemberOrTraverse(GroupRepositoryProvider.java:158)
	at org.carlspring.strongbox.providers.repository.GroupRepositoryProvider.resolvePathTraversal(GroupRepositoryProvider.java:110)
	at org.carlspring.strongbox.providers.repository.GroupRepositoryProvider.resolvePath(GroupRepositoryProvider.java:319)
	at org.carlspring.strongbox.providers.repository.GroupRepositoryProvider.resolvePath(GroupRepositoryProvider.java:35)
	at org.carlspring.strongbox.services.impl.ArtifactResolutionServiceImpl.resolvePath(ArtifactResolutionServiceImpl.java:132)
	at org.carlspring.strongbox.services.ArtifactManagementService.getPath(ArtifactManagementService.java:387)
	at org.carlspring.strongbox.services.ArtifactManagementService$$FastClassBySpringCGLIB$$b290f309.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:747)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
	at org.carlspring.strongbox.services.ArtifactManagementService$$EnhancerBySpringCGLIB$$6fd15f58.getPath(<generated>)
	at org.carlspring.strongbox.controllers.BaseArtifactController.provideArtifactDownloadResponse(BaseArtifactController.java:274)
	at org.carlspring.strongbox.controllers.maven.MavenArtifactController.download(MavenArtifactController.java:144)
	at org.carlspring.strongbox.controllers.maven.MavenArtifactController$$FastClassBySpringCGLIB$$ce982786.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:747)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.security.access.intercept.aopalliance.MethodSecurityInterceptor.invoke(MethodSecurityInterceptor.java:69)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
	at org.carlspring.strongbox.controllers.maven.MavenArtifactController$$EnhancerBySpringCGLIB$$7042ec8c.download(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:209)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:136)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:102)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:871)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:777)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:991)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:925)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:978)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:870)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:855)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:320)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
	at org.carlspring.strongbox.security.authentication.CustomAnonymousAuthenticationFilter.doFilter(CustomAnonymousAuthenticationFilter.java:43)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.carlspring.strongbox.security.authentication.StrongboxAuthenticationFilter.doFilterInternal(StrongboxAuthenticationFilter.java:61)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.carlspring.strongbox.security.authentication.JwtTokenValidationFilter.doFilter(JwtTokenValidationFilter.java:57)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:96)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:64)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:357)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:270)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.carlspring.strongbox.web.HeaderMappingFilter.doFilter(HeaderMappingFilter.java:118)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:357)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:270)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.carlspring.strongbox.web.DirectoryTraversalFilter.doFilterInternal(DirectoryTraversalFilter.java:33)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:564)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:317)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:110)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:128)
	at org.eclipse.jetty.util.thread.Invocable$InvocableExecutor.invoke(Invocable.java:222)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:294)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:199)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:673)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:591)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.FileNotFoundException: The artifact path is a directory: [/strongbox/strongbox-vault/storages/storage-common-proxies/carlspring/.index]
	at org.carlspring.strongbox.providers.layout.RepositoryLayoutFileSystemProvider.newOutputStream(RepositoryLayoutFileSystemProvider.java:151)
	at java.nio.file.Files.newOutputStream(Files.java:216)
	at org.carlspring.strongbox.providers.repository.ProxyRepositoryProvider.getOutputStream(ProxyRepositoryProvider.java:96)
	at org.carlspring.strongbox.services.impl.ArtifactResolutionServiceImpl.getOutputStream(ArtifactResolutionServiceImpl.java:88)
	at org.carlspring.strongbox.services.ArtifactManagementService.store(ArtifactManagementService.java:116)
	... 143 common frames omitted
22:42:07.755 28-01-2018 | DEBUG | qtp1914190643-20     | o.c.s.providers.repository.GroupRepositoryProvider | Located artifact: [/strongbox/strongbox-vault/storages/storage-common-proxies/carlspring/.index]
22:42:07.755 28-01-2018 | DEBUG | qtp1914190643-20     | o.c.s.providers.repository.GroupRepositoryProvider | Located artifact: [/strongbox/strongbox-vault/storages/storage-common-proxies/carlspring/.index]
22:42:07.755 28-01-2018 | DEBUG | qtp1914190643-20     | o.c.s.controllers.maven.MavenArtifactController    | Resolved path : /strongbox/strongbox-vault/storages/storage-common-proxies/carlspring/.index
``

**Comment by carlspring** [comment]
@sanket407 ,

Is this perhaps something you could look into?

**Comment by sanket407** [comment]
@carlspring  What is  the request for?
If is it .index file then browsing logic currently doesnt support for files starting with "."

**Comment by carlspring** [comment]
@sanket407 , I was hoping your recent fixes on #557 (#548) would solve this, but I've just checked it against a Maven 2 proxy repository and, unfortunately, this issue is still present.

Maven 2 proxy repositories have two types of indexes a `local` and `remote` one. These directories are created for them, even if they're empty when the server starts. Currently, it doesn't seem to be possible to browse these for proxy repositories.

Would you mind having a look a this as when you can?

Thanks! :)

**Comment by carlspring** [comment]
Sorry, it seems I tested it against an incorrect revision! This works fine! :)

**Comment by carlspring** [comment]
Solved with #557 .
